### DISCUSSION TITLE
Get the actual item when passed to the callback

### DISCUSSION THREAD

**Discussion by Mte90** [zauberzeug/nicegui#4021] [Ideas / Feature Requests]
In this code:
```python
_id = -1
                for client in self.clients:
                    with ui.card().classes('items-center p-4 shadow-md rounded cursor-pointer') as customer_card:
                        _id += 1
                        self.element['customer_card_' + str(_id)] = customer_card
                        ui.image(client["avatar"]).classes('w-16 h-16 rounded-full mb-2')
                        ui.label(client["name"]).classes('text-lg')
                        ui.button("Elimina", on_click=lambda: self.delete_customer(id)).classes(
                            'bg-blue-500 text-white mt-2')
```

I get as `id` always `4` that is the last item in the loop also if I click on the first item.
It seems like that the callback doesn't get the value associated when is added but at the end itself, so I was looking to a way to get the element that got the click.
Maybe like a `this` in javascript (I will need to get the parent but a problem at time).

There is any suggestion on how to do that or it is missing something from nicegui itself?

**Comment by Mte90** [comment]
I fixed it, it required a parameter for the lambda function.

**Reply by falkoschindler** [reply]
Exactly!
[Why do all my elements have the same value?](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-do-all-my-elements-have-the-same-value)
